### ISSUE TITLE
noUndeclaredVariables complains about default-exported const in .d.ts file

### ISSUE THREAD

**Issue by printfn** [biomejs/biome#2637]
### Environment information

```bash
CLI:
  Version:                      1.7.1
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.11.1"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "npm/10.2.4"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           true
  Linter disabled:              false
  Organize imports disabled:    true
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```

### Rule name

lint/correctness/noUndeclaredVariables

### Playground link

https://biomejs.dev/playground/?lintRules=all&code=ZABlAGMAbABhAHIAZQAgAG0AbwBkAHUAbABlACAAJwB0AGUAcwB0AC0AbQBvAGQAdQBsAGUAJwAgAHsACgAgACAALwAvACAAYgBpAG8AbQBlAC0AaQBnAG4AbwByAGUAIABsAGkAbgB0AC8AcwB1AHMAcABpAGMAaQBvAHUAcwAvAG4AbwBFAHgAcABsAGkAYwBpAHQAQQBuAHkAOgAKACAAIABjAG8AbgBzAHQAIABmAG8AbwA6ACAAYQBuAHkAOwAKACAAIAAvAC8AIABiAGkAbwBtAGUALQBpAGcAbgBvAHIAZQAgAGwAaQBuAHQALwBzAHQAeQBsAGUALwBuAG8ARABlAGYAYQB1AGwAdABFAHgAcABvAHIAdAA6AAoAIAAgAGUAeABwAG8AcgB0ACAAZABlAGYAYQB1AGwAdAAgAGYAbwBvADsACgB9AAoA&jsx=false

### Expected result

It should not error.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct
